lib/puppet/type/archive.rb

```diff
@@ -174,7 +174,12 @@ def should_to_s(value)
   end
 
   newparam(:checksum_url) do
-    desc 'archive file checksum source (instead of specifying checksum)'
+    desc 'archive file checksum source (instead of specifying checksum). Supports http|https|ftp|file'
+    validate do |value|
+      unless value =~ URI.regexp(%w[http https ftp file])
+        raise ArgumentError, "invalid source url: #{value}"
```

Comment:
thanks for adding the validation, could you add a test here? https://github.com/voxpupuli/puppet-archive/blob/master/spec/unit/puppet/type/archive_spec.rb